### PR TITLE
Mention Git Dependancy in Documention

### PR DESCRIPTION
**Pull request by yashexe**
## Description
PR to resolve the [Git Dependancy Issue (#3571)](https://github.com/kedro-org/kedro/issues/3571). 

## Development notes
The following markdown files have been slightly modified to mention Git Dependancy:
- docs\source\course\index.md
- docs\source\deployment\single_machine.md
- docs\source\faq\faq.md
- docs\source\get_started\new_project.md
- docs\source\starters\create_a_starter.md

I havent been able to rebuild the documentation as I am a Windows user, but I used the "Markdown All in One" extension on VSCode to verify that it renders properly. I've read [that this is allowed](https://github.com/kedro-org/kedro/wiki/Contribute-to-the-Kedro-documentation#how-do-i-rebuild-the-documentation-after-i-make-changes-to-it), but please let me know if you would like me to go about it in a different way.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Hi @yashexe, thanks a lot for your contribution! We'll have a look at it shortly 🙏🏼

File: docs/source/course/index.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ We assume you know these concepts:
 
 We don't assume knowledge of software engineering in Python, so the course contains information about reusability principles, how to create a Python package, and how to use version control.
 
+Please note that we do expect users to have Git installed, as it is a prerequisite for the kedro-new flow, which is used when starting a new project.
```
Comment (merelcht) [inline docs/source/course/index.md:25]:
```suggestion
Please note that we do expect users to have Git installed, as it is a prerequisite for the `kedro new` flow, which is used when creating a new project.
```

**Review by merelcht — APPROVED** [review]
One small suggestion, but otherwise looks good! Thanks for the contribution @yashexe ⭐

**Review by ElenaKhaustova — APPROVED** [review]
Thank you, @yashexe!

**Comment by yashexe** [comment]
Thanks for the approval guys! Hope to contribute more in the near future!

**Pull request merged**